packages/react-router/modules/useRouter.js

```diff
@@ -0,0 +1,25 @@
+import { useContext, version as REACT_VERSION } from "react";
+import invariant from "tiny-invariant";
+
+import RouterContext from "./RouterContext";
+
+// needed for ESM builds
+let useRouter = null;
+
+if (useContext) {
+  useRouter = function useRouter() {
+    const context = useContext(RouterContext);
+    invariant(context, "You should not use useRouter() outside a <Router>");
```

Comment:
Maybe `"You should not use 'useRouter()' or other hooks outside a <Router>"` ?

Since the other hooks use this one internally, the error message as is might become a red herring in some scenarios. Like the "Don't use <Route/>"-warning has been for some people (including me, tbh).

Reply:
I noticed the issue by myself, but could not find a good solution to solve it. I think your proposal is also not the best solution, but definitely better than the current message. Will change it.

Reply:
Done.